lib/ansible/modules/remote_management/manageiq/manageiq_group.py

```diff
@@ -0,0 +1,601 @@
+                    self.module.fail_json(msg="Multiple tenants found in manageiq with name '%s" % tenant_name)
+                tenant = tenant_res[0]
+                return tenant
+            else:
+                # No tenant name or tenant id supplied
+                return None
+
+    def role(self, role_id, role_name):
+        """ Search for a role object by name or id.
+        Returns:
+            the role entity, None no id or name was supplied
+
+            the role, or send a module Fail signal if role not found.
+        """
+        if role_id:
+            role = self.client.get_entity('roles', role_id)
+            if not role:
+                self.module.fail_json(msg="Role with id '%s' not found in manageiq" % str(role_id))
+            return role
+        else:
+            if role_name:
+                role_res = self.client.collections.roles.find_by(name=role_name)
+                if not role_res:
+                    self.module.fail_json(msg="Role '%s' not found in manageiq" % role_name)
+                if len(role_res) > 1:
+                    self.module.fail_json(msg="Multiple roles found in manageiq with name '%s" % role_name)
+                return role_res[0]
+            else:
+                # No role name or role id supplied
+                return None
+
+    @staticmethod
+    def have_same_values_nested_array(current_values, updated_values):
+        if (not current_values) and (not updated_values):
+            return True
+        if current_values and (not updated_values):
+            return False
+        if (not current_values) and updated_values:
+            return False
+        if len(current_values) != len(updated_values):
+            return False
+
+        # sort subs
+        for sub in current_values:
+            sub.sort()
+        for sub in updated_values:
+            sub.sort()
+
+        current_values.sort()
+        updated_values.sort()
+
+        return current_values == updated_values
+
+    @staticmethod
+    def merge_values_nested_array(current_values, updated_values):
+        if current_values and (not updated_values):
+            return current_values
+        if (not current_values) and updated_values:
+            return updated_values
+
+        current_values_categories = {}
+        new_values_categories = {}
+
+        # organize the values per category name
+        for sub in current_values:
+            sub.sort()
+            key = sub[0].split('/')[2]
+            current_values_categories[key] = sub
+
+        for sub in updated_values:
+            sub.sort()
+            key = sub[0].split('/')[2]
+            new_values_categories[key] = sub
+
+        # add and update current categories
+        for key in new_values_categories:
+            current_values_categories[key] = new_values_categories[key]
+
+        res = []
+
+        for key in current_values_categories:
+            res.append(current_values_categories[key])
```

Comment:
I think this can be simplified to `return current_values_categories.values()`.

Reply:
Yes. for python3 is must be return list(current_values_categories.values()), still cleaner.